lib/modules.nix

```diff
@@ -342,28 +347,52 @@ rec {
       };
 
       result = withWarnings {
+        _type = "configuration";
         options = checked options;
         config = checked (removeAttrs config [ "_module" ]);
         _module = checked (config._module);
         inherit extendModules type;
       };
     in result;
 
-  # collectModules :: (modulesPath: String) -> (modules: [ Module ]) -> (args: Attrs) -> [ Module ]
+  # collectModules :: (class: String) -> (modulesPath: String) -> (modules: [ Module ]) -> (args: Attrs) -> [ Module ]
   #
   # Collects all modules recursively through `import` statements, filtering out
   # all modules in disabledModules.
-  collectModules = let
+  collectModules = class: let
```

Comment:
This is technically a breaking change since `collectModule` is exposed under `lib.modules.collectModule`, but that's arguably a design mistake, it should never have been exposed, I'd be in favor of making it or marking it internal somehow.

Reply:
The time has come to deprecate the functions we consider to be probably internal. By deprecating them, we can figure out whether anyone uses them, while hopefully relieving us of this supposedly needless contract soon-ish.